File: solutions/clojure/log-levels/1/src/log_levels.clj
(ns log-levels
  (:require [clojure.string :as str]))

(defn message
  "Takes a string representing a log line
   and returns its message with whitespace trimmed."
  [s]
  (str/trim(second(str/split (str/trim-newline s) #": " 2))))

(defn log-level
  "Takes a string representing a log line
   and returns its level in lower-case."
  [s]
  (str/lower-case(second(str/split s #"[\]\[]"))))

(defn reformat
  "Takes a string representing a log line and formats it
   with the message first and the log level in parentheses."
  [s]
  (format "%s (%s)" (message s) (log-level s)))

File: solutions/java/log-levels/1/src/main/java/LogLevels.java
public class LogLevels {

    private static String logLevel;
    private static String logMessage;

    private static void parseLogLine(String logLine) {
        String[] string = logLine.split("]:", 2);
        logLevel        = string[0].replace("[", "");
        logMessage      = string[1].trim();
    }

    public static String message(String logLine) {
        parseLogLine(logLine);
        return logMessage;
    }

    public static String logLevel(String logLine) {
        parseLogLine(logLine);
        return logLevel.toLowerCase();
    }

    public static String reformat(String logLine) {
        parseLogLine(logLine);
        return "".format("%s (%s)", logMessage, logLevel.toLowerCase());
    }
}
